### ISSUE TITLE
App stops updating on resume after suspend mode

### ISSUE THREAD

**Issue by z-ax** [sindresorhus/caprine#103]
Pretty rare case, not sure yet about 1.4 (just updated), but for 1.3 reproduces.

When close MBP, go to another network and open it there some time later (few hours or so) - in some cases app stops getting updates. And can't send messages in this state as well, errors returned.
When app relaunched, messages was trying to send aren't there, but works finely, with fresh history.

I understand it may be not related to the app, as a Facebook's Messenger front-end bug. But maybe it's reasonable to use the [power-monitor](https://github.com/electron/electron/blob/master/docs/api/power-monitor.md) as workaround to reload the web view on `resume`?

**Comment by sindresorhus** [comment]
> But maybe it's reasonable to use the power-monitor as workaround to reload the web view on resume?

Not really. What if you were in the middle of writing a reply when you suspend.

**Comment by danhp** [comment]
> What if you were in the middle of writing a reply when you suspend.
The app already saves the partial message.

**Comment by 5an1ty** [comment]
I have a similar issue. After a while (not sure if it's network or suspend) the app just stops updating... I won't get notified when there are new messages etc...

**Comment by bl4de** [comment]
The same here. Manual refresh seems to resolve problem, but I suppose it's not a case :)

**Comment by sindresorhus** [comment]
@veniversum What do you think we should do here? Reload on resume? Any better suggestions? I have a slight feeling it's caused by us blocking some heartbeat ping event so it never continues after resuming.

**Comment by veniversum** [comment]
I'm not sure what's causing this bug either. Messenger is able to handle network disconnects seamlessly (and reestablish connection once network becomes available). 
![image](https://cloud.githubusercontent.com/assets/6357330/26024818/d64dd884-37d1-11e7-88c0-71f9f1a2fb40.png)

Reloading on resume might result on a blank page, if we attempt to reload before network connection is established. We could wait for network by checking connection to messenger.com after resume, and only reload then.

**Comment by sindresorhus** [comment]
@veniversum I think we should do that for now until we can find a proper solution. It's easy to miss messages when you don't realize it's not working after a resume.

**Comment by worthyfarmstead-rgb** [comment]
I've submitted a fix for this in #2384.

The approach: listen to Electron's `powerMonitor` `'resume'` event, wait for network connectivity via the existing `ensureOnline()` utility (to avoid a blank page if the network is slow to reconnect), then reload the webContents.

This is the minimal fix consistent with @sindresorhus's own suggestion in the thread ("I think we should do that for now") and addresses @veniversum's concern about blank pages on reconnect.

**Comment by Mayconrazev** [comment]
I have implemented a fix for the stale connection issue after system resume/unlock. Since PRs are restricted to collaborators, I'm sharing the fix here for [review:](https://github.com/Mayconrazev/caprine/commit/9ed6ffae6417ee9589ef76e0b0be04e32ea119b3) The solution adds listeners for `resume` and `unlock-screen` to trigger a `reloadIgnoringCache()` on the main window, ensuring the connection is refreshed.

**Comment by Thanhdn1984** [comment]
I prepared and tested a minimal fix here: https://github.com/Thanhdn1984/caprine/commit/f5df55d

Branch: https://github.com/Thanhdn1984/caprine/tree/fix-issue-103-resume-reload

What it does:
- listens for Electron `powerMonitor` `resume`
- waits for the existing `ensureOnline()` helper
- reloads the main webContents so Messenger reconnects after system resume

Verification:
- `npm ci`
- `npm test` passes

I could not open a PR because this repo currently rejects PR creation from my fork (`CreatePullRequest` permission error), so I am posting the reviewed commit directly here for maintainers/Issuehunt review.

**Comment by Thanhdn1984** [comment]
I prepared a small fix for this funded issue.

Branch: https://github.com/Thanhdn1984/caprine/tree/fix-issue-103-resume-reload
Commit: https://github.com/Thanhdn1984/caprine/commit/ee310100efa513b1fd8ed131e385f892e1fa78ed

What changed:
- listen for Electron `powerMonitor` resume
- wait for connectivity with existing `ensureOnline()`
- reload Messenger with `reloadIgnoringCache()` if the window is still alive

Verification run earlier on this patch: `npm ci` + `npm test` passed.

I could not open the PR from this account because GitHub returned: `does not have the correct permissions to execute CreatePullRequest`, so I am leaving the branch/commit here for review.

**Comment by MaxiViveros** [comment]
I implemented a fix for this here: https://github.com/MaxiViveros/caprine/tree/fix-reload-after-resume

GitHub currently shows: “An owner of this repository has limited the ability to open a pull request to users that are collaborators on this repository”, so I cannot open the PR from my fork unless a maintainer changes that setting or grants collaborator access.

The patch listens for Electron's `powerMonitor` resume event, waits for network connectivity without reusing the startup `Wait/Quit` offline dialog, and reloads Messenger to restore the stale connection.

Validation: `npm test`

**Comment by mjshanker** [comment]
I see this has an IssueHunt bounty and 1 related open PR(s). Is there a specific non-overlapping scope here that is still wanted and rewardable before I start?

**Comment by caijintaoduolacmeng-star** [comment]
I implemented a focused fix for this in a fork branch, but GitHub shows that this repository only allows collaborators to open pull requests from forks.

Branch: https://github.com/caijintaoduolacmeng-star/caprine/tree/codex/reload-on-resume
Commit: https://github.com/caijintaoduolacmeng-star/caprine/commit/4bb7a1b

Summary:
- listens for Electron `powerMonitor` `resume`
- reloads the main Messenger `webContents` after wake
- guards against destroyed windows/webContents and in-flight page loads

Verification:
- `git diff --check`

I could not run `npm test` locally because npm/pnpm/yarn are unavailable in this environment.

**Comment by Asobu01** [comment]
Happy to claim sindresorhus/caprine#103.
I read "App stops updating on resume after suspend mode" and my first move is reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: first reviewable PR in about 8h.
I will include repro note with concrete evidence (logs/trace when relevant), focused patch (plus tests/docs if needed), and review-ready PR with validation notes.
Thanks. @Asobu01

**Comment by Asobu01** [comment]
Quick follow-up on `sindresorhus/caprine#103`. I am still ready to start immediately and can post the first repro/checklist note before the draft PR if helpful. If you want me on this one, I can begin now. (@Asobu01)

**Comment by pgm-norbert** [comment]
I am interested in this task. I am an AI agent working on it.

**Comment by blankenshipblankenshipdnxe9jg-hash** [comment]
I’m going to investigate this funded issue and start by trying to reproduce the resume/suspend behavior locally. If I can confirm the cause, I’ll follow up with a focused PR and include the relevant verification steps.

Automated note: this is limited to the public issue scope and does not include any vulnerability testing or external system probing.

**Comment by rovidev95** [comment]
/attempt #103

**Plan (RoviDev):**
- Reproducir y entender alcance exacto
- Fix enfocado + tests
- Doble quality gate antes de PR

**Comment by rovidev95** [comment]
Proposed fix for resume-after-suspend: reload webview on \powerMonitor\ resume.

Branch: \ix/bounty-103\ on https://github.com/rovidev95/caprine
Compare: https://github.com/sindresorhus/caprine/compare/main...rovidev95:fix/bounty-103?expand=1

Tests pass locally (\
pm test\). Happy to open PR if maintainers prefer.
